android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/ssdp/Ssdp.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2014 Fraunhofer FOKUS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * AUTHOR: Louay Bassbouss <louay.bassbouss@fokus.fraunhofer.de>
+ *
+ */
+package org.physical_web.physicalweb.ssdp;
+
+import android.util.Log;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetSocketAddress;
+import java.net.MulticastSocket;
+import java.net.SocketAddress;
+import java.net.SocketTimeoutException;
+
+/**
+ * Created by lba on 29.01.2015.
+ */
+public class Ssdp implements Runnable {
+    public static String TAG = "SSDP";
+    public static String SSDP_ADDRESS = "239.255.255.250";
+    public static int SSDP_PORT = 1900;
+    public static int SSDP_SOURCE_PORT = 1901;
+    public static String SSDP_HOST = SSDP_ADDRESS + ":" + SSDP_PORT;
+    public static String MAX_AGE = "max-age=1800";
+    public static int TTL = 128;
+    public static int MX = 2;
+    public static String ALIVE = "ssdp:alive";
+    public static String BYEBYE = "ssdp:byebye";
+    public static String UPDATE = "ssdp:update";
+    public static String TYPE_M_SEARCH = "M-SEARCH";
+    public static String TYPE_NOTIFY = "NOTIFY";
+    public static String TYPE_200_OK = "200 OK";
+
+    private SsdpCallback mSsdpCallback;
+    private SocketAddress mMulticastGroup;
+    //private MulticastSocket mMulticastSocket;
+    private DatagramSocket mMulticastSocket;
```

Comment:
Instead of mMulticastSocket, can we use mDatagramSocket, since MulticastSocket is actually another class?